File: ios/AllEasyTest/Modules/SystemVersionModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE (SystemVersionModule, NSObject)
RCT_EXTERN_METHOD(getSystemVersion
                  : (RCTPromiseResolveBlock)resolve rejecter
                  : (RCTPromiseRejectBlock)reject)
@end

File: ios/AllEasyTest/Modules/SystemVersionModule.swift
import Foundation
import UIKit

@objc(SystemVersionModule)
class SystemVersionModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }

  @objc
  func getSystemVersion(
    _ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    let version = UIDevice.current.systemVersion
    resolve(version)
  }
}
